android_vs_ios.md

```diff
@@ -0,0 +1,175 @@
+
+Modern Android layouts are typically done using [ConstraintLayout](https://developer.android.com/reference/android/support/constraint/ConstraintLayout), which is very similar to iOS's Auto Layout. [LinearLayout](https://developer.android.com/guide/topics/ui/layout/linear) and [FrameLayout](https://developer.android.com/reference/android/widget/FrameLayout) are occasionally used for simpler views, and [RelativeLayouts](https://developer.android.com/guide/topics/ui/layout/relative) were used on older projects as it is deprecated by ConstraintLayout. A Layout is just a container that dictates how its children are positioned. Each layout file will have one of these layouts as the top level container, and it is possible (and quite common) for a layout to contain another layout.\
+Historically, layouts are generally built by writing xml instead of using the interface builder, partly due to Android Studio's interface builder being terrible/inadequate. The interface builder has improved quite a lot recently with the introduction of ConstraintLayout, so it has more usage nowadays. However, building the layout via by xml might still be more preferable to a lot of people since it offers more control, and Android Studio can help autocomplete many of the fields.
+
+## Libraries
+[Gradle](https://developer.android.com/studio/build/), the official build system for Android, includes the ability to manage libraries and dependencies.\
+iOS projects generally use [CocoaPods](https://cocoapods.org/) for managing libraries. [Swift Package Manager](https://swift.org/package-manager/) is the official dependency manager for Swift that was released somewhat recently, although Cocoapods is still more widely used.
+
+Due to slow/nonexistent updates from Android device manufacturers, many Android devices are stuck on older version of Android. This makes it very difficult to develop for as certain features might only be available on newer version Android and certain method calls behave differently on different OS versions. Google provides [Android Support Library](https://developer.android.com/topic/libraries/support-library/) that backports certain new features to older OS version, provides consistent behavior across different OS version, and generally abstracts away a lot of boilerplate when dealing with backwards compatibility. **It is highly recommended to use the support library calls/classes instead of platform version whenever possible.** Additionally, Google recently introduces [Architecture Components](https://developer.android.com/topic/libraries/architecture/) that tries solve some common headaches in Android development such as dealing with lifecycle and database.\
+There are no equivalent of these on iOS.
```

Comment:
Maybe a sentence or two on why there's no need on iOS. Faster updates and forced obsolescence of older OS version means that new APIs trickle down much faster and deprecated things don't need to be supported indefinitely.